cdap-api/src/main/java/co/cask/cdap/internal/schedule/constraint/Constraint.java

```diff
@@ -19,5 +19,5 @@
 /**
  * A constraint that is checked before executing a schedule.
  */
-public abstract class Constraint {
+public interface Constraint {
```

Comment:
the implementation of the Constraints need to extend from a class in `cdap-app-fabric`, the `AbstractConstraint`, so had to make this one just an interface.